pkg/elastic_service/elastic_handler.go

```diff
@@ -0,0 +1,147 @@
+package elastic_service
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/daniyakubov/book_service_n/pkg/book_service/models"
+	"github.com/daniyakubov/book_service_n/pkg/consts"
+	errors "github.com/fiverr/go_errors"
+	"github.com/olivere/elastic/v7"
+)
+
+type ElasticHandler struct {
+	Ctx          *context.Context
+	Url          string
+	Client       *elastic.Client
+	maxSizeQuery int
+}
+
+func NewElasticHandler(ctx *context.Context, url string, client *elastic.Client, maxSizeQuery int) ElasticHandler {
+	return ElasticHandler{
+		ctx,
+		url,
+		client,
+		maxSizeQuery,
+	}
+}
+
+func (e *ElasticHandler) Post(title string, id string) (err error) {
+	_, err = e.Client.Update().
+		Index(consts.Index).
+		Id(id).
+		Doc(map[string]interface{}{consts.Title: title}).
+		Do(*e.Ctx)
+	if err != nil {
+		return errors.Wrap(err, err.Error())
+	}
+	return nil
+}
+
+func (e *ElasticHandler) Put(postBody []byte) (string, error) {
+	put, err := e.Client.Index().
+		Index(consts.Index).
+		BodyString(string(postBody)).
+		Do(*e.Ctx)
+	if err != nil {
+		return "", errors.Wrap(err, err.Error())
+	}
+	return put.Id, err
+}
+func (e *ElasticHandler) Get(id string) (src *models.Book, err error) {
+	get, err := e.Client.Get().
+		Index(consts.Index).
+		Id(id).
+		Do(*e.Ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, err.Error())
+	}
+
+	var getResp models.Book
+	if err = json.Unmarshal(get.Source, &getResp); err != nil {
+		return nil, err
+	}
+	return &getResp, err
+}
+
+func (e *ElasticHandler) Delete(id string) error {
+	_, err := e.Client.Delete().
+		Index(consts.Index).
+		Id(id).
+		Do(*e.Ctx)
+	if err != nil {
+		return errors.Wrap(err, err.Error())
+	}
+	return err
+}
+
+func (e *ElasticHandler) Search(title string, author string, priceStart float64, priceEnd float64) (res []models.Book, err error) {
+	s := ""
+	if title == "" && author == "" && priceEnd == 0 {
```

Comment:
the if-else pattern is not a common way to write in go (sometimes it is necessary..) please try to think of a better solution. furthermore, the "price_range" that the user is inserting is one string (you can define the format for it), please make sure that there are min and max ranges etc